sim-cli/src/main.rs

```diff
@@ -60,7 +61,8 @@ async fn main() -> anyhow::Result<()> {
         .unwrap();
 
     let SimParams { nodes, activity } =
-        serde_json::from_str(&std::fs::read_to_string(cli.sim_file)?)?;
+        serde_json::from_str(&std::fs::read_to_string(cli.sim_file)?)
+            .map_err(|e| anyhow!("Cannot deserialize node connection data into LndConnection nor ClnConnection (line {}, col {})", e.line(), e.column()))?;
```

Comment:
I just realised this may not be good, given this can error out both for node parsing and activity parsing

Reply:
Maybe just make the error more holistic? 
`Could not deserialize node connection (LND or CLN) or activity description: (line {}, col {}) in simulation file.`

Most folks should be able to figure it out based on the line). Perhaps also include the error string if there is one? Since it's often `hanging ,` or some other annoying json error.

Reply:
I like your version better.

Regarding including the original string, there's actually no meaningful info there, that's why I decided to go with a custom error. 

```
Error: data did not match any variant of untagged enum NodeConnection at line 11 column 8
```

Not even the line and column are that useful, they mark the end of the section where the error is at, not even the real line :/

Reply:
Hm ok, let's just get as user friendly as we can and ship it.